cmd/thv-operator/pkg/git/client.go

```diff
@@ -125,19 +151,39 @@ func (*DefaultGitClient) GetFileContent(repoInfo *RepositoryInfo, path string) (
 }
 
 // Cleanup removes local repository directory
-func (*DefaultGitClient) Cleanup(repoInfo *RepositoryInfo) error {
+func (*DefaultGitClient) Cleanup(ctx context.Context, repoInfo *RepositoryInfo) error {
+	logger := log.FromContext(ctx)
 	if repoInfo == nil || repoInfo.Repository == nil {
-		return nil
+		return fmt.Errorf("repository is nil")
 	}
 
-	// Get the repository directory from the worktree
-	workTree, err := repoInfo.Repository.Worktree()
-	if err != nil {
-		return fmt.Errorf("failed to get worktree: %w", err)
+	// 1. Clear object cache explicitly
+	if repoInfo.objectCache != nil {
+		logger.V(1).Info("Clearing object cache")
+		repoInfo.objectCache.Clear()
+	}
+
+	// 2. Clear worktree filesystem
+	worktree, err := repoInfo.Repository.Worktree()
+	if err == nil && worktree.Filesystem != nil {
+		logger.V(1).Info("Clearing worktree filesystem")
+		_ = util.RemoveAll(worktree.Filesystem, "/")
 	}
 
-	// Remove the directory
-	return os.RemoveAll(workTree.Filesystem.Root())
+	// 3. Clear storer filesystem (memfs)
+	if repoInfo.storerFilesystem != nil {
+		logger.V(1).Info("Clearing storer filesystem")
+		_ = util.RemoveAll(repoInfo.storerFilesystem, "/")
+	}
+
+	// 4. Nil out all references
+	repoInfo.objectCache = nil
+	repoInfo.storerFilesystem = nil
+	repoInfo.Repository = nil
+
+	// 5. Force GC to reclaim memory
+	runtime.GC()
```

Comment:
I still don't think nudging the GC is necessary.

Reply:
Testing it